### PR TITLE
Link state of mission control buttons

### LINKED ISSUE #663: Connect MissionControlButtons in OngoingMissionCard and MissionPage
**Describe the improvement you would like to see**
MissionControlButtons in OngoingMissionCard and inside the MissionPage should be connected.

**How will this change existing functionality?**
MissionControlButtons in OngoingMissionCard and inside MissionPage will have the same state. This prevents edge cases where the buttons can be pressed rapidly in series by entering or leaving MissionPage and re-click them before the state is updated, leaving the buttons in "wait state".

**How will this improvement affect the current Threat Model?**
N/A

### PR DESCRIPTION
**Pull request by andchiind**
Closes #663

### CODE REVIEW

File: flotilla.code-workspace
Comment (phhuseboe) [inline flotilla.code-workspace:1]: I think this file can be removed

File: frontend/src/components/Contexts/MissionControlContext.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { createContext, useContext, useState, FC } from 'react'
+import { BackendAPICaller } from 'api/ApiCaller'
+import { Mission } from 'models/Mission'
+
+export enum ControlButton {
+    Pause,
+    Stop,
+    Resume,
+}
```
Comment (phhuseboe) [inline frontend/src/components/Contexts/MissionControlContext.tsx:9]: ControlButton is already defined in MissionControlButtons.tsx, so we could consider to import it from there

File: frontend/src/components/Contexts/MissionControlContext.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const setIsWaitingForResponse = (isWaiting: boolean) => {
+        setMissionControlState({ isWaitingForResponse: isWaiting })
+    }
+
+    const handleClick = (button: ControlButton, mission: Mission) => {
+        switch (button) {
+            case ControlButton.Pause: {
+                setIsWaitingForResponse(true)
+                BackendAPICaller.pauseMission(mission.robot.id).then((_) => setIsWaitingForResponse(false))
+                //.catch((e) => handleError(e))
```
Comment (Christdej) [inline frontend/src/components/Contexts/MissionControlContext.tsx:43]: Even if you just move the function, I think we can remove the comment

File: frontend/src/components/Contexts/MissionControlContext.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                break
+            }
+            case ControlButton.Resume: {
+                setIsWaitingForResponse(true)
+                BackendAPICaller.resumeMission(mission.robot.id).then((_) => setIsWaitingForResponse(false))
+                //.catch((e) => handleError(e))
```
Comment (Christdej) [inline frontend/src/components/Contexts/MissionControlContext.tsx:49]: Even if you just move the function, I think we can remove the comment

File: frontend/src/components/Contexts/MissionControlContext.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                break
+            }
+            case ControlButton.Stop: {
+                setIsWaitingForResponse(true)
+                BackendAPICaller.stopMission(mission.robot.id).then((_) => setIsWaitingForResponse(false))
+                //.catch((e) => handleError(e))
```
Comment (Christdej) [inline frontend/src/components/Contexts/MissionControlContext.tsx:55]: Even if you just move the function, I think we can remove the comment

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -4,22 +4,25 @@ import { FlotillaSite } from 'components/Pages/FlotillaSite'
...
-                        <SignInPage></SignInPage>
-                    </div>
-                </UnauthenticatedTemplate>
-                <AuthenticatedTemplate>
-                    <FlotillaSite />
-                </AuthenticatedTemplate>
-            </>
-            {/* </ErrorBoundary> */}
+            <MissionControlProvider>
+                {/* // <ErrorBoundary fallbackRender={({ error, resetErrorBoundary }) => ErrorFallback(error)}> */}
```
Comment (Christdej) [inline frontend/src/App.tsx:13]: I think we can remove the comment

**Review by Christdej — CHANGES_REQUESTED** [review]

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**